client/web/src/enterprise/campaigns/apply/CampaignSpecMissingCredentialsAlert.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import { Link } from '../../../../../shared/src/components/Link'
+import { pluralize } from '../../../../../shared/src/util/strings'
+import { AuthenticatedUser } from '../../../auth'
+import { defaultExternalServices } from '../../../components/externalServices/externalServices'
+import { ViewerCampaignsCodeHostsFields } from '../../../graphql-operations'
+
+export interface CampaignSpecMissingCredentialsAlertProps {
+    viewerCampaignsCodeHosts: ViewerCampaignsCodeHostsFields
+    authenticatedUser: Pick<AuthenticatedUser, 'url'>
+}
+
+export const CampaignSpecMissingCredentialsAlert: React.FunctionComponent<CampaignSpecMissingCredentialsAlertProps> = ({
+    viewerCampaignsCodeHosts,
+    authenticatedUser,
+}) => {
+    if (viewerCampaignsCodeHosts.totalCount === 0) {
+        return <></>
+    }
+    return (
+        <div className="alert alert-warning">
+            <h4 className="alert-heading">
+                You don't have credentials configured for{' '}
+                {pluralize('this code host', viewerCampaignsCodeHosts.totalCount, 'these code hosts')}
+            </h4>
```

Comment:
I think the `h4` with `alert-heading` looks weird. This looks better, IMHO:

```suggestion
        <div className="alert alert-warning">
            <p>
                <b>
                    You don't have credentials configured for{' '}
                    {pluralize('this code host', viewerCampaignsCodeHosts.totalCount, 'these code hosts')}:
                </b>
            </p>
```